### PR TITLE
Valet Admin Component

### PR DESCRIPTION
**Pull request by bzw24**
### Valet Admin Component

1. Validate that the user role is 'admin'. Render out the menu accordingly to admin or user
2. Valet Admin displays a card that contains parking garage details of the admin user
3. Stats of Number of Spots
4. Stats of Operation Hours
5. When clicking edit spots or hours, a modal opens up with a form prepopulate to edit fields
6. Controller handles all cases for errors and does checks in the backend, error message populates accordingly with toasts as well
7. Separated up the modals into from main component

Future refactor: 

- handle input errors on the front end as well (backend will already handle for all situations, but its good to have client end fail safes as well)

image

image

image

image

### CODE REVIEW

File: src/components/account/AccountSetting.jsx [hunk 1]
```diff
@@ -29,12 +29,15 @@ const AccountSetting = () => {
     fetchUserData();
   }, [dispatch, id]);
 
-  const handlePhoneNumberChange = (e) => {
-    const input = e.target.value;
-    const formattedInput = formatPhoneNumber(input);
-    setPhoneNumber(formattedInput);
-    dispatch(setUserDataPhoneNumber(formattedInput));
-  };
+  const handlePhoneNumberChange = useCallback(
```
Comment (jose-m-f) [inline src/components/account/AccountSetting.jsx:32]: This is a nice optimization with usecallback

File: src/components/account/EditParkingSpotsModal.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            e.preventDefault();
+            handleParkingSpotsUpdate();
+          }}
+        >
+          <label className="form-control w-full max-w-xs">
+            <div className="label">
+              <span className="label-text">Edit Number of Parking Spots</span>
+            </div>
+            <input
+              type="text"
```
Comment (jose-m-f) [inline src/components/account/EditParkingSpotsModal.jsx]: I think this input element could be type="number" to easily prevent unexpected values

**Review by jose-m-f — APPROVED** [review]
I left a couple of comments, one is a change suggestion that you could make in a future PR. Other than that it, looks great!

File: src/components/account/EditParkingSpotsModal.jsx (hunk 2 above)
Comment (bzw24) [reply src/components/account/EditParkingSpotsModal.jsx]: Good catch, change it to number type

**Review by stevkim — APPROVED** [review]
Looks great for functionality! Great work bruce!

**Comment by bzw24** [comment]
> Looks great for functionality! Great work bruce!



> Looks great for functionality! Great work bruce!

edited the export accountsSlice reducer to follow same suit as example.

**Pull request merged**